tests/test_tomography_gate_set_tomography.py

```diff
@@ -0,0 +1,401 @@
+            prepared_observable = _get_observable(j, nqubits)
+    else:
+        prepared_observable = _get_observable(j, nqubits).form
+        groundtruth = correct_observables[nqubits][j]
+        assert groundtruth == prepared_observable
+
+
+@pytest.mark.parametrize("nqubits", range(3, 7))
+def test_expectation_value(backend, nqubits):
+
+    if nqubits == 3:
+        test_circuit = qibo.models.Circuit(nqubits)
+        test_circuit.add(gates.TOFFOLI(0, 1, 2))
+        test_circuit.add(gates.M(*np.arange(0, nqubits, 1)))
+
+    else:
+        RandUnitary = RU(2**nqubits)
+        test_circuit = qibo.models.Circuit(nqubits)
+        test_circuit.add(gates.Unitary(RandUnitary, *np.arange(0, nqubits, 1)))
+        test_circuit.add(gates.M(*np.arange(0, nqubits, 1)))
+    k = 1
+    j = 1
+    with pytest.raises(ValueError):
+        expectation_val = _expectation_value(
+            test_circuit,
+            j,
+            nshots=int(1e4),
+            backend=backend,
+        )
+
+
+@pytest.mark.parametrize("nqubits", range(1, 4))
+@pytest.mark.parametrize("gate", [gates.CNOT(0, 1), gates.TOFFOLI(0, 1, 2), None])
+def test_gate_set_tomography(backend, nqubits, gate):
+    test_circuit = qibo.models.Circuit(nqubits)
+    test_circuit.add(gates.H(0))
+
+    errors = {
+        (1, gates.CNOT(0, 1)): ValueError,
+        (1, gates.TOFFOLI(0, 1, 2)): ValueError,
+        (2, gates.TOFFOLI(0, 1, 2)): ValueError,
+        (3, gates.TOFFOLI(0, 1, 2)): ValueError,
+    }
+
+    if (nqubits, gate) in [
+        (1, gates.CNOT(0, 1)),
+        (1, gates.TOFFOLI(0, 1, 2)),
+        (2, gates.TOFFOLI(0, 1, 2)),
+        (3, gates.TOFFOLI(0, 1, 2)),
+    ]:
+        with pytest.raises(ValueError):
+            matrix_jk = _gate_set_tomography(
+                nqubits=nqubits,
+                gate=test_gate,
+                nshots=int(1e4),
+                invert_register=None,
+                noise_model=None,
+                backend=backend,
+            )
+
+
+def test_gate_set_tomography_invert_register(backend):
+    nqubits = 1
+    test_circuit = qibo.models.Circuit(nqubits)
+    test_circuit.add(gates.H(0))
+    matrix_jk = _gate_set_tomography(
+        nqubits=nqubits,
+        gate=None,
+        nshots=int(1e6),
+        invert_register=(0,),
+        noise_model=None,
+        backend=backend,
+    )
+    groundtruth = np.array([[1, 1, 1, 1], [0, 0, 0, 0], [0, 0, 0, 0], [1, 1, 1, 1]])
+    backend.assert_allclose(
+        matrix_jk,
+        groundtruth,
+        atol=1e-2,
+    )
+
+
+def test_reset_register_valid_tuple_1qb():
+    # Test for valid tuple
+    nqubits = 1
+    test_circuit = qibo.models.Circuit(nqubits)
+    test_circuit.add(gates.H(0))
+
+    invert_register = (0,)
+    inverse_circuit = reset_register(test_circuit, invert_register)
+
+    correct_gates = [
+        [gates.H(0)],
+    ]
+    for groundtruth, gate in zip(correct_gates, inverse_circuit.queue):
+        assert isinstance(gate, type(groundtruth[0]))
```

Comment:
As suggested above I would remove the invert circuit for now and, thus, all the associated tests.